graylog2-server/src/main/java/org/graylog2/shared/rest/resources/ProxiedResource.java

```diff
@@ -365,6 +364,45 @@ protected <RemoteInterfaceType, RemoteCallResponseType> NodeResponse<RemoteCallR
         return requestOnLeader(remoteInterfaceFunction, interfaceClass, getDefaultProxyCallTimeout());
     }
 
+    /**
+     * Execute the given remote interface function on the given node.
+     * <p>
+     * This is used to forward an API request to the given node. It is useful in situations where an API call has to
+     * be executed on a particular, chosen node.
+     * <p>
+     * The returned {@link NodeResponse} object is constructed from the remote response's status code and body.
+     */
+    protected <RemoteInterfaceType, RemoteCallResponseType> NodeResponse<RemoteCallResponseType> requestOnNode(
+            String nodeId,
+            Function<RemoteInterfaceType, Call<RemoteCallResponseType>> remoteInterfaceFunction,
+            Class<RemoteInterfaceType> interfaceClass,
+            Duration timeout
+    ) throws IOException {
+
+        final Node executionNode = nodeService.allActive().values().stream()
+                .filter(node -> Objects.equals(nodeId, node.getNodeId()))
+                .findFirst()
+                .orElseThrow(() -> new IllegalStateException(String.format(Locale.ENGLISH, "Node %s cannot be found among active nodes", nodeId)));
```

Comment:
You can use our `StringUtils.f()` method to avoid passing the locale. :slightly_smiling_face: